warehouse/packaging/metadata_verification.py

```diff
@@ -41,6 +56,109 @@ def _verify_url_pypi(url: str, project_name: str, project_normalized_name: str)
     )
 
 
+def _get_url_content(
+    resolved_ip: str, url: rfc3986.api.URIReference, max_length_bytes: int
+) -> bytes:
+    """
+    Get at least `max_length_bytes` of the contents of a URL
+
+    Raises `GetContentError` on any errors while trying to
+    access the URL.
+    """
+    try:
+        http = urllib3.HTTPSConnectionPool(
+            host=resolved_ip,
+            port=443,
+            headers={
+                "Host": url.host,
+                "User-Agent": "warehouse/1.0 (PyPI URL verifier, https://pypi.org)",
+            },
+            server_hostname=url.host,
+            assert_hostname=url.host,
+            retries=False,
+            timeout=5,
+        )
+        r = http.request("GET", url.path if url.path else "/", preload_content=False)
+    except Exception:
+        raise GetContentError()
+
+    content = r.read(max_length_bytes)
+
+    # When using `preload_content=False`, the HTTP connection must be manually closed
+    r.close()
+
+    return content
+
+
+def _verify_url_meta_tag(
+    url: str, project_name: str, project_normalized_name: str
+) -> bool:
+    user_uri = rfc3986.api.uri_reference(url).normalize()
```

Comment:
To avoid URL parser differentials we should parse using `urllib3.util.parse_url()`. We can cross-check the result of the `scheme` and `host` and `port`.